crates/brainfuck_prover/src/components/mod.rs

```diff
@@ -21,3 +26,73 @@ pub enum TraceError {
     #[error("The trace is empty.")]
     EmptyTrace,
 }
+
+/// Represents the different trace types used in the Brainfuck STARK proving system.
+#[derive(Debug, Eq, PartialEq)]
+pub enum Trace {
+    /// Memory access trace.
+    Memory,
+    /// Instruction execution trace.
+    Instruction,
+    /// Input/output trace.
+    Io,
+    /// Processor trace (register).
+    Processor,
+}
+
+impl Trace {
+    /// Returns the number of columns associated with the specific trace type.
+    pub fn column_count(&self) -> usize {
+        match self {
+            Self::Memory => MemoryColumn::count(),
+            Self::Instruction => InstructionColumn::count(),
+            Self::Io => IoColumn::count(),
+            Self::Processor => unimplemented!(),
+        }
+    }
+}
+
+/// Represents a claim associated with a specific trace in the Brainfuck STARK proving system.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Claim {
+    /// Logarithmic size (`log2`) of the evaluated trace.
+    pub log_size: u32,
+    /// Type of the associated trace.
+    pub trace: Trace,
```

Comment:
Not a big fan of having a dedicated field for the component type of the Claim

Would be great if it can be abstracted